clientconn.go

```diff
@@ -1740,8 +1748,13 @@ func (cc *ClientConn) parseTargetAndFindResolver() error {
 	// We are here because the user's dial target did not contain a scheme or
 	// specified an unregistered scheme. We should fallback to the default
 	// scheme, except when a custom dialer is specified in which case, we should
-	// always use passthrough scheme.
-	defScheme := resolver.GetDefaultScheme()
+	// always use passthrough scheme. For either case, we need to respect any overridden
+	// global defaults set by the user.
+	defScheme := cc.dopts.defScheme
+	if defScheme == "" || internal.UserSetDefaultScheme {
```

Comment:
`defscheme` will never be `""` with these changes, since you always set it.

I think it's OK to just delete the `defScheme == ""` part.  (We should use our default default unless the user set a different default.)